### PR TITLE
feat(typescript_indexer): add support for ref/writes

### PR DESCRIPTION
**Pull request by wcalandro**
Adds support for `ref/writes` edges to the Typescript indexer. The verifier test is based on the recent test made for the Java indexer.

### CODE REVIEW

**Review by jaysachs — COMMENTED** [review]
How does this behave for
```
arr[x]++
```
?

File: kythe/typescript/testdata/writes.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+//- @#1z ref Z
+//- !{ @#1z ref/writes Z }
+z += z;
+
+//- @arr defines/binding Arr
+const arr = [1, 2, 3];
+
+//- @arr ref Arr
+//- !{ @arr ref/writes Arr }
+arr[0] = 1;
```
Comment (jaysachs) [inline kythe/typescript/testdata/writes.ts:174]: For completeness, how about a test of
```
arr[0]++;
```

**Review by jaysachs — APPROVED** [review]

File: kythe/typescript/testdata/writes.ts (hunk 1 above)
Comment (wcalandro) [reply kythe/typescript/testdata/writes.ts]: Ah I had written the test already but forgot to push the commit. Fixed now

File: kythe/typescript/testdata/writes.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+//- @arr ref Arr
+//- !{ @arr ref/writes Arr }
+arr[0]++;
```
Comment (schroederc) [inline kythe/typescript/testdata/writes.ts:178]: Add trailing newline to file

File: kythe/typescript/testdata/writes.ts [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+//- @#1z ref Z
+//- !{ @#1z ref/writes Z }
+z += z;
+
+//- @arr defines/binding Arr
+const arr = [1, 2, 3];
+
+//- @arr ref Arr
+//- !{ @arr ref/writes Arr }
+arr[0] = 1;
```
Comment (schroederc) [inline kythe/typescript/testdata/writes.ts:174]: Remove these checks since the behavior is still under debate

**Review by schroederc — CHANGES_REQUESTED** [review]

**Review by schroederc — APPROVED** [review]

**Pull request merged**